### DISCUSSION TITLE
Autofill Starting Position

### DISCUSSION THREAD

**Discussion by st11x** [inkstitch/inkstitch#2038] [Q&A]
Hi,

In this example, I manually set the starting position to the middle but when stitched out it always starts at the point circled. This happens to some objects and because of that, it makes a jump to the starting position. 

![image](https://user-images.githubusercontent.com/7398499/215355684-f40080cb-2acb-4f08-a052-e16db5e979ec.png)

Here's a short clip of the stitch plan where it starts by jumping to the starting position.

https://user-images.githubusercontent.com/7398499/215355709-89598e01-990e-447d-a7f9-054f7746f610.mp4

Any ideas? 

thanks
   Matt

**Comment by lexelby** [comment]
I suspect there might be a super-tiny shape right at that spot where it's starting.  If a fill object has multiple shapes in it, Ink/Stitch will pick one to start with arbitrarily.

@kaalleen Maybe we should ignore tiny fill subpaths too?

**Reply by st11x** [reply]
Someone on Reddit told me to troubleshoot those objects and I typically end up with "border crosses itself" even when it is obviously not. However, if I move it slightly, copy and paste it back, or do something along those lines, I can fix those problems and once that is resolved, it seems to fix this as well.

**Reply by lexelby** [reply]
Ah!  It's likely correct that the border does cross itself, but only on a super-tiny scale.  Imagine the shape of the number 8, except one of the loops is microscopic while the other is normal size.

When Ink/Stitch sees a border crossing like that, it tries to cope with it by splitting it into two shapes and sewing them separately.  That's likely what's happening here.

Why does moving it fix the problem?  A couple of possibilities.  It might be that when you move it, Ink/Stitch chooses to sew the main shape first before the microscopic shape.  Or maybe the microscopic shape of small enough to fit _between_ the rows of stitches.  Or maybe the border doesn't cross itself anymore!  When you're dealing with things on a microscopic scale like that in a computer, things get weird.  It's kind of like quantum scale effects in the tell world :)

Inkscape's Simplify command might help.  Or you could try deleting path nodes in the area that's causing problems.

@kaalleen Maybe when we repair a shape we should discard tiny shapes?  Do we already try to do that?

**Comment by kaalleen** [comment]
Yes, seems to make sense. At the moment we just make sure it doesn't break the rendering.
For now the break apart fill objects extension should be able to repair that.

**Reply by st11x** [reply]
Thanks. I was using the wrong break apart and now that I found the one under Ink/Stitch -> Tools : Fill, that has helped resolve most of the issues.
